src/rantanplan/rhymes.py

```diff
@@ -255,8 +261,18 @@ def analyze_rhyme(lines, offset=4):
                 stressed_endings, assonance, relaxation, offset
             )
             rhyme = "".join(rhymes)
-            rhythmical_lengths = [line["rhythm"]["length"] for line in lines]
-            ranking = search_structure(rhyme, rhythmical_lengths, rhyme_type)
+            # rhythmical_lengths = [line["rhythm"]["length"] for line in lines]
+            length_ranges = [
+                range(line["rhythm"]["length_range"]["min_length"],
+                      line["rhythm"]["length_range"]["max_length"] + 1)
+                for line in lines]
+            candidates = search_structure(rhyme, length_ranges, rhyme_type)
+
+            if candidates is not None:
+                ranking, *_ = candidates
+            else:
+                ranking = None
+
```

Comment:
This is what I mean by making the function always return a list, albeit empty in some cases.
```suggestion
            candidates = search_structure(rhyme, length_ranges, rhyme_type)
            if len(candidates):
                ranking, *_ = candidates
            else:
                ranking = None
```

Reply:
Yes you're right, it's done